app/models/program.rb

```diff
@@ -41,6 +41,7 @@ def confirmed
 
   before_create :create_event_types
   before_create :create_difficulty_levels
+  before_save :check_languages_format
```

Comment:
Why a before_save hook? What happens if it fails? Does the user get an error message? For me this looks more like a validation ...